packages/terra-worklist-data-grid/src/subcomponents/ColumnHeaderCell.jsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import '../_elementPolyfill';
+import * as KeyCode from 'keycode-js';
+import styles from '../WorklistDataGrid.module.scss';
+import WorklistDataGridUtils from '../utils/WorklistDataGridUtils';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * String identifier of the column in which the Cell will be rendered.
+   */
+  columnId: PropTypes.string.isRequired,
+  coordinates: PropTypes.shape({
+    x: PropTypes.number,
+    y: PropTypes.number,
+  }),
+  /**
+   * Boolean value to indicate if the cell can accept focus.
+   */
+  acceptsFocus: PropTypes.bool,
+  width: PropTypes.string,
+  height: PropTypes.string,
+  onMoveCellFocus: PropTypes.func,
+  onCellSelectionChange: PropTypes.func,
+  /**
+   * Content that will rendered within the Cell.
+   */
+  children: PropTypes.node,
+};
+
+const defaultProps = {
+  acceptsFocus: false,
+};
+
+function ColumnHeaderCell(props) {
+  const {
+    columnId,
+    coordinates,
+    acceptsFocus,
+    width,
+    height,
+    onMoveCellFocus,
+    onCellSelectionChange,
+    children,
+    ...customProps
+  } = props;
+
+  const handleClick = (event) => {
+    if (onCellSelectionChange) {
+      onCellSelectionChange(null, null, coordinates);
+    }
+
+    event.preventDefault();
+  };
+
+  const handleKeyDown = (event) => {
```

Comment:
The WorklistDataGrid should be handling this business logic.  The cell would just have to report back the new row and column index.  The cell does not need to know about the row and column lengths.  The WorklistDataGrid has that info.

Reply:
Was there any followup to this? Is there are reason why keyboard navigation is down within this component and not at the root level?

CC: @brithom

Reply:
Question was answered offline. +1